include/deal.II/base/exceptions.h

```diff
@@ -1519,6 +1519,49 @@ namespace deal_II_exceptions
 } /*namespace deal_II_exceptions*/
 
 
+#if defined(__clang__)
+#  define DEAL_II_ASSUME(expr)                     \
+    do                                             \
+      {                                            \
+        __builtin_assume(static_cast<bool>(expr)); \
+      }                                            \
+    while (false)
+#elif defined(__GNUC__) && !defined(__ICC)
+#  if __GNUC__ >= 13
+#    define DEAL_II_ASSUME(expr)                                         \
+      do                                                                 \
+        {                                                                \
+          _Pragma("GCC diagnostic push")                                 \
+            _Pragma("GCC diagnostic ignored \"-Wimplicit-fallthrough\"") \
+              [[assume(expr)]];                                          \
```

Comment:
Is `[[assume]]` available unconditionally with GCC version >= 13, even though you're only in C++17 mode?

Reply:
Yes, seems that way. I didn't request a particular C++ standard support.

Reply:
OK, let's give it a try!